### ISSUE TITLE
How to get started on macOS? Multiple approaches have failed.

### ISSUE THREAD

**Issue by zbeekman** [icecc/icecream#476]
If anyone could point me in the right direction getting icecream setup for the first time on macOS it would be much appreciated.

## TL;DR: Not setting `ICECC_VERSION` causes everything to build locally, creating the environment and setting `ICECC_VERSION` gives: `xcode-select: error: no developer tools were found at '/Applications/Xcode.app'...`

I have two machines on my network, an i9 MPB w/ 12 threads/6 phys cores and an iMac with 4 cores. Bother are running `iceccd` and `icecc-scheduler` and the only options set are log-location, verbosity, and `iceccd` sets a hostname and the number of cores.

I'm trying to setup icecream on macOS to split expensive builds between multiple macOS machines on the local network. My first question is, am I compiling on the remote machine at all? The answer appears (to me) to be "no".

```
$  env | grep C[CX]
CC=/usr/local/Cellar/icecream/1.2_1/libexec/icecc/bin/cc
ICECC_DEBUG=debug
MPICH_CXX=/usr/local/opt/gcc@8/bin/g++-8
MPICH_CC=/usr/local/opt/gcc@8/bin/gcc-8
$ echo $PATH
/usr/local/opt/icecream/libexec/icecc/bin:/usr/local/sbin:/usr/local/opt/jenv/shims:/usr/local/bin
```

When I configure a project with CMake all seems well, but while building I see everything is run locally:
```
[ 93%] Building C object programs/CMakeFiles/mpmetis.dir/stat.c.o
ICECC[20232] 2019-06-19 18:56:22: invoked as: /usr/local/Cellar/icecream/1.2_1/libexec/icecc/bin/cc -DHAVE_RINT -DSVNINFO="" -I/Users/<user>/Sandbox/<PROJECT>/submodules/metis/GKlib -I/Users/<user>/Sandbox/<PROJECT>/submodules/metis/include -I/Users/<user>/Sandbox/<PROJECT>/submodules/metis/programs/../libmetis -DLINUX -D_FILE_OFFSET_BITS=64 -DNDEBUG -DNDEBUG2 -DHAVE_EXECINFO_H -DHAVE_GETLINE -O3 -o CMakeFiles/mpmetis.dir/stat.c.o -c /Users/<user>/Sandbox/<PROJECT>/submodules/metis/programs/stat.c
ICECC[20232] 2019-06-19 18:56:22: connected to /var/run/icecc/iceccd.socket
ICECC[20232] 2019-06-19 18:56:22: native /var/tmp/icecc-envs/native/
ICECC[20232] 2019-06-19 18:56:22: env: Darwin18_x86_64 '/var/tmp/icecc-envs/native/'
ICECC[20232] 2019-06-19 18:56:22: preparing /Users/<user>/Sandbox/<PROJECT>/submodules/metis/programs/stat.c to be compiled for Darwin18_x86_64
ICECC[20232] 2019-06-19 18:56:22: $ICECC_VERSION needs to point to .tar files
ICECC[20232] 2019-06-19 18:56:22: got exception Error 22 - $ICECC_VERSION needs to point to .tar files (this should be an exception!)
ICECC[20232] 2019-06-19 18:56:22: connected to /var/run/icecc/iceccd.socket
ICECC[20232] 2019-06-19 18:56:22: <building_local>
ICECC[20179] 2019-06-19 18:56:22: </building_local: 21708ms>
ICECC[20184] 2019-06-19 18:56:22: invoking: /usr/bin/cc -DHAVE_RINT -DSVNINFO="" -I/Users/<user>/Sandbox/<PROJECT>/submodules/metis/GKlib -I/Users/<user>/Sandbox/<PROJECT>/submodules/metis/include -I/Users/<user>/Sandbox/<PROJECT>/submodules/metis/programs/../libmetis -DLINUX -D_FILE_OFFSET_BITS=64 -DNDEBUG -DNDEBUG2 -DHAVE_EXECINFO_H -DHAVE_GETLINE -O3 -c /Users/<user>/Sandbox/<PROJECT>/submodules/metis/programs/cmdline_gpmetis.c -o CMakeFiles/gpmetis.dir/cmdline_gpmetis.c.o
ICECC[20174] 2019-06-19 18:56:22: </building_local: 21755ms>
```

If I explicitly create a tarball with `icecc --build-native` and then set `ICECC_VERSION=/Users/<user>/Sandbox/x86_64-macos-10.14-clang.tar.gz` I get errors:
```
    xcode-select: error: no developer tools were found at '/Applications/Xcode.app', and no install could be requested (perhaps no UI is present), please install manually from 'developer.apple.com'.
```

Is there something I'm missing to successfully package the Xcode toolchain?

I've attached the tarball here: https://www.dropbox.com/s/9v4gn2y1bh3wdvt/x86_64-macos-10.14-clang.tar.gz?dl=0 (It's 41MB)

Any ideas for how to get up and running with icecream would be much appreciated.

**Comment by torarnv** [comment]
Did you install the Xcode 11 beta? It messed up the command line tools for me. Try installing the command line tools for Xcode 10 that you download from Apple's website.

**Comment by timblechmann** [comment]
iirc i had to specify the compiler explicitly (though i'm running ccache/icecc via cmake's compiler launcher). i've documented my findings: https://timblechmann.gitlab.io/posts/2019-06-10-icecream.html

**Comment by llunak** [comment]
In the first case, when you let icecream create the tarball itself, then according to the log message it seems to me that icecc-create-env fails, but I don't know why.

**Comment by llunak** [comment]
Is there still something to do here? You can try the current development version of Icecream as it has better logging when it comes to handling the environment. As for the Mac-specific parts, I don't know if there's a way to help you more than the link posted above.

**Comment by zbeekman** [comment]
Hi sorry, I haven't had time to circle back to this. I'll try to look at it this weekend. Thanks!

**Comment by permcody** [comment]
Hi @zbeekman. Based on your description, it sounds like you need to install either XCode or the Command line tools. One of these is necessary to get started on Mac OS. Take a look at our getting started for working with raw compilers on MOOSE:
https://mooseframework.inl.gov/getting_started/installation/macos_catalina.html

**Comment by cinakyn** [comment]
I also have troubled with same issue.
In my case, the problem was missing `icecc` binary on the worker machine.
